quickwit/quickwit-cli/src/index.rs

```diff
@@ -643,122 +645,199 @@ pub async fn describe_index_cli(args: DescribeIndexArgs) -> anyhow::Result<()> {
     let metastore = metastore_uri_resolver
         .resolve(&quickwit_config.metastore_uri)
         .await?;
-    let index_metadata = metastore.index_metadata(&args.index_id).await?;
-    let splits = metastore
-        .list_splits(&args.index_id, SplitState::Published, None, None)
-        .await?;
+    let index_stats = IndexStats::from_metastore(metastore, &args.index_id).await?;
```

Comment:
I'd get the index metadata and splits outside of this function and then change the signature to `IndexStats::new(index_metadata, splits)`.